core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/TwitterDao.java

```diff
@@ -0,0 +1,149 @@
+package ca.jrvs.apps.twitter.dao;
+
+import ca.jrvs.apps.twitter.dao.helper.HttpHelper;
+import ca.jrvs.apps.twitter.util.JsonParser;
+import ca.jrvs.apps.twitter.model.Tweet;
+import com.google.gdata.util.common.base.PercentEscaper;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+public class TwitterDao implements CrdDao<Tweet, String> {
+
+    //URI constants
+    private static final String API_BASE_URI = "https://api.twitter.com";
+    private static final String POST_PATH = "/1.1/statuses/update.json";
+    private static final String SHOW_PATH = "/1.1/statuses/show.json";
+    private static final String DELETE_PATH = "/1.1/statuses/destroy.json";
+
+    //URI symbols
+    private static final String QUERY_SYM = "?";
+    private static final String AMPERSAND = "&";
+    private static final String EQUAL = "=";
+
+    //Response code
+    private static final int HTTP_OK = 200;
```

Comment:
use [this](http://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpStatus.html)